Source/WebCore/page/Navigation.cpp

```diff
@@ -316,4 +317,62 @@ bool Navigation::hasEntriesAndEventsDisabled() const
     return false;
 }
 
+static NavigationNavigationType determineNavigationType(FrameLoadType frameLoadType)
+{
+    switch (frameLoadType) {
+    case FrameLoadType::Standard:
+        return NavigationNavigationType::Push;
+    case FrameLoadType::Back:
+    case FrameLoadType::Forward:
+    case FrameLoadType::IndexedBackForward:
+        return NavigationNavigationType::Traverse;
+    case FrameLoadType::Reload:
+    case FrameLoadType::ReloadFromOrigin:
+    case FrameLoadType::ReloadExpiredOnly:
+    case FrameLoadType::Same:
+        return NavigationNavigationType::Reload;
+    case FrameLoadType::RedirectWithLockedBackForwardList:
+    case FrameLoadType::Replace:
+        return NavigationNavigationType::Replace;
+    };
+    RELEASE_ASSERT_NOT_REACHED();
+    return NavigationNavigationType::Push;
+}
+
+// https://html.spec.whatwg.org/multipage/nav-history-apis.html#update-the-navigation-api-entries-for-a-same-document-navigation
+void Navigation::updateForNavigation(Ref<HistoryItem>& item, FrameLoadType frameLoadType)
+{
+    if (hasEntriesAndEventsDisabled())
+        return;
+
+    auto oldCurrentNHE = currentEntry();
```

Comment:
I don't think we should use the abbreviation NHE.  It is quite unclear what it means.

Reply:
I followed the specification, but indeed they are a bit cryptic, fixed.